tools/flow_grid.py

```diff
@@ -0,0 +1,689 @@
+import numpy as np
+import pandas as pd
+import sys
+import ast
+
+class flow_grid(object):        
```

Comment:
There are a few places in this script that have some extra whitespace at the end of lines and between lines, can you trim those?